cgstats.js

```diff
@@ -126,37 +126,74 @@ app.get('/search*', function(req, res) {
       });
     });
   } else if (optimizations.indexOf(game) != -1) {
+    getStats('LeaderboardsRemoteService/getFilteredPuzzleLeaderboard',
+      [game, , , {"active" : true, "column" : "keyword", "filter" : player}])
+      .then(function (response) {
```

Comment:
this response contains rankings only for requested keyword, like

```
{ users: 
   [ { pseudo: 'd-s-x',
       rank: 63,
       globalRank: 63,
       localRank: 63,
       score: 100,
       testSessionHandle: '48277672d595cdc4b03e02abca6bda19f64a5cb',
       programmingLanguage: 'Bash',
       inProgress: false,
       criteriaScore: 38,
       company: '',
       codingamer: [Object] },
     { pseudo: 'd-s-x',
       rank: 64,
       globalRank: 64,
       localRank: 64,
       score: 100,
       testSessionHandle: '48277672d595cdc4b03e02abca6bda19f64a5cb',
       programmingLanguage: 'Perl',
       inProgress: false,
       criteriaScore: 38,
       company: '',
       codingamer: [Object] },
     { pseudo: 'd-s-x',
       rank: 221,
       globalRank: 221,
       localRank: 221,
       score: 100,
       testSessionHandle: '48277672d595cdc4b03e02abca6bda19f64a5cb',
       programmingLanguage: 'Ruby',
       inProgress: false,
       criteriaScore: 50,
       company: '',
       codingamer: [Object] },
...
```

so we have to download more stats but with language filter